partiql-planner/src/testFixtures/resources/inputs/basics/coalesce.sql

```diff
@@ -0,0 +1,71 @@
+--#[coalesce-00]
+-- type: (int32)
+COALESCE(1);
+
+--#[coalesce-01]
+-- type: (int32)
+COALESCE(1, 2);
+
+--#[coalesce-02]
+-- type: (decimal)
+COALESCE(1, 1.23);
+
+--#[coalesce-03]
+-- type: (null | decimal)
+COALESCE(NULL, 1, 1.23);
+
+--#[coalesce-04]
+-- type: (null | missing | decimal)
+COALESCE(NULL, MISSING, 1, 1.23);
+
+--#[coalesce-05]
+-- type: (null | missing | decimal); same as above
+COALESCE(1, 1.23, NULL, MISSING);
+
+--#[coalesce-06]
+-- type: (int32)
+COALESCE(t_item.t_int32);
+
+--#[coalesce-07]
+-- type: (int32)
+COALESCE(t_item.t_int32, t_item.t_int32);
+
+--#[coalesce-08]
+-- type: (int64)
+COALESCE(t_item.t_int64, t_item.t_int32);
+
+--#[coalesce-09]
+-- type: (int64 | null)
+COALESCE(t_item.t_int64_null, t_item.t_int32, t_item.t_int32_null);
+
+--#[coalesce-10]
+-- type: (int64 | null | missing)
+COALESCE(t_item.t_int64_null, t_item.t_int32, t_item.t_int32_null, MISSING);
+
+--#[coalesce-11]
+-- type: (int64 | string)
+COALESCE(t_item.t_int64, t_item.t_string);
+
+--#[coalesce-12]
+-- type: (int64 | null | string)
+COALESCE(t_item.t_int64_null, t_item.t_string);
+
+--#[coalesce-13]
+-- type: (int16 | int32 | int64 | int | decimal)
+COALESCE(t_item.t_num_exact, t_item.t_int32);
```

Comment:
Would this not be `type: decimal`?

Reply:
The union of all the numeric types is output here to follow the existing `CASE WHEN` behavior. I believe this was brought up in Yingtao's comment about a union type for one of the `THEN` branches https://github.com/partiql/partiql-lang-kotlin/pull/1391#pullrequestreview-1943767991. For instance, the current behavior of `CASE WHEN` will include all of the types within the union/AnyOfType:
```
CASE <type string>
    WHEN 'a' THEN <type union(int4, int8)>
    ELSE <type int4>
END;
```
will have return type of `union(int4, int8)`.

This is also the case for `COALESCE` which is equivalent to `CASE WHEN` with each argument corresponding to a `THEN` branch:
```
COALESCE(<type union(int4, int8)>, <type int4>)
==
CASE WHEN <type union(int4, int8)> IS NOT NULL THEN <type union(int4, int8)>
     ELSE <type int4>
END
```
will have return type of `union(int4, int8)`.

One could make the argument that the minimal common supertype is actually just `int8`; however, this `union` type argument behavior has not been defined yet.